Sources/BrowserServicesKit/RemoteMessaging/Model/AnyDecodable.swift

```diff
@@ -0,0 +1,183 @@
+// Copyright 2018 Read Evaluate Press, LLC
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software
+// and associated documentation files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial
+// portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+// PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+// FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+//
+// https://github.com/Flight-School/AnyCodable/blob/master/Sources/AnyCodable/AnyDecodable.swift
+
+#if canImport(Foundation)
+import Foundation
+#endif
+
+@frozen public struct AnyDecodable: Decodable {
+    public let value: Any
+
+    public init<T>(_ value: T?) {
+        self.value = value ?? ()
+    }
+}
+
+@usableFromInline
+protocol _AnyDecodable {
+    var value: Any { get }
+    init<T>(_ value: T?)
+}
+
+extension AnyDecodable: _AnyDecodable {}
```

Comment:
What is this file used for?

Reply:
@amddg44 can correct me if I'm wrong, but I believe this facilitates parsing of dictionaries like the `translations` while still using a `JSONDecoder()` 

See https://github.com/duckduckgo/BrowserServicesKit/pull/125/files#diff-55a677ad8562e792bf8f1384c1f39effbf7947777399f208653996959d2f0bb5R17

Would have been better imo if it were a array of translation objects with the local and as attribute, then map that to a dictionary in memory after it was parsed, but don't think we can fix that that now.

Reply:
The reason for AnyDecodable is primarily based upon decoding the rules with its various structures. I guess the other approach is to change the iOS json file structure in a way that AnyDecodable wouldn't be needed - however this could potentially create a significant amount of work for whoever is creating the json files. The process would no longer be start with one platform, then copy for second platform and do a couple of platform specific tweaks.

https://github.com/duckduckgo/BrowserServicesKit/blob/0c0b8114f8121ecf0886d7ec2f7f2e11fc3f683a/Tests/BrowserServicesKitTests/Resources/remote-messaging-config.json#L102